File: CS/dxSample_ColoredFormHeaders/Colored - header form/CustomFormPainter.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using DevExpress.Skins;
using DevExpress.Skins.XtraForm;
using DevExpress.Utils.Drawing;

namespace dxSample_ColoredFormHeaders {
    public class CustomFormPainter : FormPainter {
        public CustomFormPainter(System.Windows.Forms.Control owner, DevExpress.Skins.ISkinProvider provider)
            : base(owner, provider) {
        }
        private Color GetFormBorderColor() {
            var formBorderColor = (Owner as ColoredHeaderForm).FormBorderColor;
            return formBorderColor;
        }
        protected override void DrawBackground(GraphicsCache cache) {
            var info = GetCaptionInfo();
            var ee = info as ObjectInfoArgs;
            var formBorderColor = GetFormBorderColor();
            cache.FillRectangle(new SolidBrush(formBorderColor), ee.Bounds);
        }
        protected override void DrawFrameCore(GraphicsCache cache, SkinElementInfo info, FrameKind kind) {
            var formBorderColor = GetFormBorderColor();
            cache.FillRectangle(formBorderColor, info.Bounds);
        }
    }
}

File: VB/dxSample_ColoredFormHeaders/Colored - header form/CustomFormPainter.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports DevExpress.Skins
Imports DevExpress.Skins.XtraForm
Imports DevExpress.Utils.Drawing

Namespace dxSample_ColoredFormHeaders
    Public Class CustomFormPainter
        Inherits FormPainter

        Public Sub New(ByVal owner As System.Windows.Forms.Control, ByVal provider As DevExpress.Skins.ISkinProvider)
            MyBase.New(owner, provider)
        End Sub
        Private Function GetFormBorderColor() As Color
            Dim formBorderColor = (TryCast(Owner, ColoredHeaderForm)).FormBorderColor
            Return formBorderColor
        End Function
        Protected Overrides Sub DrawBackground(ByVal cache As GraphicsCache)
            Dim info = GetCaptionInfo()
            Dim ee = TryCast(info, ObjectInfoArgs)
            Dim formBorderColor = GetFormBorderColor()
            cache.FillRectangle(New SolidBrush(formBorderColor), ee.Bounds)
        End Sub
        Protected Overrides Sub DrawFrameCore(ByVal cache As GraphicsCache, ByVal info As SkinElementInfo, ByVal kind As FrameKind)
            Dim formBorderColor = GetFormBorderColor()
            cache.FillRectangle(formBorderColor, info.Bounds)
        End Sub
    End Class
End Namespace
